File: 20190420_TPBC2019/C-stones.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int N;
    string S;
    cin >> N;
    cin >> S;
    int left_black = 0, right_white = count(S.begin(), S.end(), '.'), min_change = left_black + right_white;
    for (int i = 0; i < N; i++) {
        if (S[i] == '#') left_black++;
        else right_white--;
        min_change = min(min_change, left_black + right_white);
    }
    cout << min_change << endl;
    return 0;
}

File: 20190420_TPBC2019/C-Stones.py
def main():
    N = int(input())
    S = input()
    left_black = 0
    right_white = S.count('.')
    min_change = left_black + right_white
    for s in S:
        if s == '#':
            left_black += 1
        else:
            right_white -= 1
        min_change = min(min_change, left_black + right_white)
    print(min_change)


if __name__ == '__main__':
    main()
